### ISSUE TITLE
Add ability to download a granule by granule name

### ISSUE THREAD

**Issue by mike-gangl** [podaac/data-subscriber#109]
As a cal/val user, I want to download the archived version of a file based on the granule name so that i can easily compare it to ones i am locally generating (same file name) and make comparisons.

```
--granule-name  -  the name of the granule to download
```
inputs:
```
collection short name
granule id
data directory
```

Note: start/stop times would not be required if this parameter is specified.

**Comment by jjmcnelis** [comment]
Suresh's edits look good to me. I'm not aware of any POCLOUD collections with files that have multiple extensions, e.g. `.nc.gz`. 

One item to look into in the future: `producer_granule_id`. I do not know how this field is used by ESDIS or PODAAC, if at all. Seems intended for cal/val use case though. https://cmr.earthdata.nasa.gov/search/site/docs/search/api.html#g-granule-ur-or-producer-granule-id We're populating it in some cases for SWOT; it's returned by default with CMR results in CSV format: 
<img width="1100" alt="image" src="https://user-images.githubusercontent.com/27017734/199843124-1c1397e0-3d05-4122-a1fb-d98835af4627.png">

**Comment by mike-gangl** [comment]
Closed with version 1.12.0

**Comment by jaffadog** [comment]
> Suresh's edits look good to me. I'm not aware of any POCLOUD collections with files that have multiple extensions, e.g. `.nc.gz`.
> 
> One item to look into in the future: `producer_granule_id`. I do not know how this field is used by ESDIS or PODAAC, if at all. Seems intended for cal/val use case though. https://cmr.earthdata.nasa.gov/search/site/docs/search/api.html#g-granule-ur-or-producer-granule-id We're populating it in some cases for SWOT; it's returned by default with CMR results in CSV format: <img alt="image" width="1100" src="https://user-images.githubusercontent.com/27017734/199843124-1c1397e0-3d05-4122-a1fb-d98835af4627.png">

unfortunately there are datasets that use .nc.gz - such as https://archive.podaac.earthdata.nasa.gov/podaac-ops-cumulus-protected/OSCAR_L4_OC_third-deg/oscar_vel11181.nc.gz. I'' raise a  bug.
